webview-ui/src/components/AnalysisPage.tsx

```diff
@@ -119,7 +109,7 @@ const AnalysisPage: React.FC = () => {
     );
   }
 
-  if (!serverRunning) {
+  if (!serverRunning && !hasViolations) {
```

Comment:
Is this just for local development purposes?

Reply:
I think we should add a follow up PR to disable server dependent actions like getSolution / runAnalysis if the server is not running while keeping this "new" use case in consideration.